lib/blockchain.go

```diff
@@ -3178,6 +3178,11 @@ func (bc *Blockchain) CreateDAOCoinLimitOrderTxn(
 	// We will set FeeNanos to it's true value after we add inputs and outputs.
 	metadata.FeeNanos = math.MaxUint64
 
+	// Initialize CancelOrderID to the ZeroBlockHash if nil.
+	if metadata.CancelOrderID == nil {
+		metadata.CancelOrderID = &ZeroBlockHash
+	}
```

Comment:
Why do we want to initialize this to the ZeroBlockHash? Doing this increases the size of the transaction